File: 30 Days of Code/Python/day_6.py
"""
Created on Tue Jul  25 12:02:55 2017

@author: Dmitry White
"""

# TDOD: Given a string, S, of length N that is indexed from 0 to N-1,
# print its even-indexed and odd-indexed characters as 2 space-separated strings on a single line.

T = int(input())
for i in range(T):
    even_s = ""
    odd_s = ""
    s = input()
    for i in filter(lambda x: x % 2 == 0, range(len(s))):
        even_s = even_s + s[i]
    for i in filter(lambda x: x % 2 != 0, range(len(s))):
        odd_s = odd_s + s[i]
    print(even_s, odd_s)

File: 30 Days of Code/JavaScript/day_6.js
/*
Created on Sat May 13 14:06 2023

@author: Dmitry White
*/

// TODO: Given a string, S, of length N that is indexed from 0 to N-1,
// print its even-indexed and odd-indexed characters as 2 space-separated strings on a single line.
// Note: 0 is considered to be an even index.

const isOdd = (n) => n % 2 !== 0;

const processLine = (line) => {
  const chars = line.split('');
  const oddChars = [];
  const evenChars = [];

  chars.forEach((char, index) => {
    isOdd(index) ? oddChars.push(char) : evenChars.push(char);
  });

  console.log(`${evenChars.join('')} ${oddChars.join('')}`);
};

function processData(input) {
  const [t, ...lines] = input.split('\n');

  for (let i = 0; i < t; i++) {
    processLine(lines[i]);
  }
}
